cspp_runner/cspp_utils.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023 Pytroll Developers
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utilities and helper functions supporting the CSPP processing runners."""
+
+import os
+import netifaces
+import logging
+import pathlib
+import shutil
+import stat
+import subprocess
+from datetime import datetime, timedelta
+from glob import glob
+
+#: Default time format
+_DEFAULT_TIME_FORMAT = '%Y-%m-%d %H:%M:%S'
+
+#: Default log format
+_DEFAULT_LOG_FORMAT = '[%(levelname)s: %(asctime)s : %(name)s] %(message)s'
+
+_NPP_SDRPROC_LOG_FILE = os.environ.get('NPP_SDRPROC_LOG_FILE', None)
+
+LOG = logging.getLogger(__name__)
+
+
+class CSPPAncillaryDataUpdater():
+    """A class to handle the update of CSPP dynamic ancillary data and LUT files."""
+
+    def __init__(self, **kwargs):
+        """Initialize the CSPP ancillary and LUTs data updater."""
+        self.url_download_trial_frequency_hours = kwargs.get('url_download_trial_frequency_hours')
+
+        self.thr_lut_files_age_days = kwargs.get('thr_lut_files_age_days')
+        self.lut_update_stampfile_prefix = kwargs.get('lut_update_stampfile_prefix')
+        self.lut_dir = kwargs.get('lut_dir')
+        self.mirror_jpss_luts = kwargs.get('mirror_jpss_luts')
+        self.url_jpss_remote_lut_dir = kwargs.get('url_jpss_remote_lut_dir')
+
+        self.url_jpss_remote_anc_dir = kwargs.get('url_jpss_remote_anc_dir')
+        self.anc_update_stampfile_prefix = kwargs.get('anc_update_stampfile_prefix')
+        self.mirror_jpss_ancillary = kwargs.get('mirror_jpss_ancillary')
+
+        self.initalize_working_dir()
+
+    def initalize_working_dir(self):
+        """Check and set the CSPP working directory."""
+        _check_environment("CSPP_WORKDIR")
+        self.cspp_workdir = os.environ.get("CSPP_WORKDIR", '')
```

Comment:
```suggestion
        self.cspp_workdir = os.environ["CSPP_WORKDIR"]
```
you're already enforcing its existence in the line above, so no need to guard and provide a default value that would anyway lead to failure in the following line